p2p/src/main/java/bisq/network/p2p/storage/persistence/SplitStoreService.java

```diff
@@ -121,39 +119,28 @@ private int parseSpecialKey(String specialKey) {
      */
     @Override
     protected void readFromResources(String postFix) {
-        // check Version.VERSION and see if we have latest r/o data store file in working directory
-        if (!new File(absolutePathOfStorageDir, getFileName() + "_" + Version.VERSION).exists())
-            makeFileFromResourceFile(postFix); // if we have the latest file, we are good, else do stuff // TODO are we?
-        else {
-            // if we have the r/o data stores in our working directory already, we can proceed on loading them.
-            File dbDir = new File(absolutePathOfStorageDir);
-            List<File> resourceFiles = Arrays.asList(dbDir.list((dir, name) -> name.startsWith(getFileName() + "_")))
-                    .stream()
-                    .map(s -> new File(s))
-                    .collect(Collectors.toList());
-
-            history = new HashMap<>();
-            store = readStore(getFileName());
-            resourceFiles.forEach(file -> {
-                SplitStore tmp = readStore(file.getName().replace(postFix, ""));
-                history.put(extractSpecialKey(file.getName(), postFix), tmp);
-            });
-        }
-    }
+        // initialize here in case this method gets called twice
+        history = new HashMap<>();
 
-    /**
-     * Extracts the special key from a file name.<br><br>
-     *
-     * Example: "TradeStatistics2Store_1.3.4_BTC_MAINNET" becomes "1.3.4"
-     *
-     * @param name the file name to begin with
-     * @param postFix the postfix eg. "BTC_MAINNET"
-     * @return the special key eg. "1.3.4"
-     */
-    private String extractSpecialKey(String name, String postFix) {
-        return name.replace(postFix, "") // remove the postfix
-                .replace("_", "") // remove the spacings
-                .replace(getFileName(), ""); // remove the file name
+        // load our live data store
+        store = readStore(getFileName());
+
+        // create file list of files we should have by now
+        List<String> versions = new ArrayList<>();
+        versions.add(Version.VERSION);
+        versions.addAll(Version.history);
+
+        // go through the list one by one
+        versions.forEach(version -> {
+            String filename = getFileName() + "_" + version + postFix; // postFix has a preceding "_"
+            if (new File(absolutePathOfStorageDir, filename).exists()) {
+                // if it is there already, load
+                history.put(version, readStore(getFileName() + "_" + version));
+            } else {
+                // either copy and split
+                history.put(version, copyAndSplit(version, postFix));
```

Comment:
> make an identical copy of the resource data and put a version number on it.

I believe you got it already. We rename the resource files and stick a "version" to it manually. So eg. with 1.3.6 we ship 3 stores: name_1.3.4_BTC_MAINNET, name_1.3.5_BTC_MAINNET, name_1.3.6_BTC_MAINNET. Then, as copyAndSplit knows about the version, each file is copied if necessary.

- the storage format remains the same. It is still Protobuf and files.
- `name_1.3.4_BTC_MAINNET` yes, exactly (I will be renaming the files in the PR once all is said and done)
- but in the future, there is not one big data store, but multiple - one for each release of Bisq
- why multiple?
  - because then we get some time relations and thus, chronological order. Once we have some order, we can start deciding which parts we keep in RAM (not in this PR, but down the line)
  - reduce the amount of data added to the repository by each release (deprecated by https://github.com/bisq-network/bisq/pull/4114)
  - in the long run, keep small chunks if possible